File: test-suite/src/test/java/io/micronaut/docs/server/routing/BackwardCompatibleController.java
package io.micronaut.docs.server.routing;

import io.micronaut.context.annotation.Requires;

// tag::imports[]
import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;
// end::imports[]
@Requires(property = "spec.name", value = "BackwardCompatibleControllerSpec")
// tag::class[]
@Controller("/hello")
public class BackwardCompatibleController {

    @Get(uris = {"/{name}", "/person/{name}"}) // <1>
    public String hello(String name) { // <2>
        return "Hello, " + name;
    }
}
// end::class[]

File: test-suite-python/src/test/python/micronaut/docs/server/routing/BackwardCompatibleController.py
from micronaut.context.annotation import Requires

# tag::imports[]
from micronaut.http.annotation import Controller, Get
# end::imports[]


@Requires(property="spec.name", value="BackwardCompatibleControllerSpec")
# tag::class[]
@Controller("/hello")
class BackwardCompatibleController:
    @Get(uris=["/{name}", "/person/{name}"])  # <1>
    def hello(self, name: str) -> str:  # <2>
        return f"Hello, {name}"
# end::class[]
